src/Shared/AppModels/AppDebug.cs

```diff
@@ -0,0 +1,229 @@
+        public AppDebug()
+        {
+            Logger.MessageReceived += Logger_MessageReceived;
+            Window.Current.SizeChanged += Current_SizeChanged;
+            MemoryManager.AppMemoryUsageIncreased += MemoryManager_MemoryChanged;
+            MemoryManager.AppMemoryUsageDecreased += MemoryManager_MemoryChanged;
+            UpdateMemoryUsage();
+            SetWindowSize();
+            _memoryWatchTimer = new DispatcherTimer();
+            _memoryWatchTimer.Interval = TimeSpan.FromSeconds(3);
+            _memoryWatchTimer.Tick += DispatchTimer_Elapsed;
+            _memoryWatchTimer.Start();
+        }
+
+        private void DispatchTimer_Elapsed(object sender, object e)
+        {
+            _memoryWatchTimer.Stop();
+            UpdateMemoryUsage();
+            _memoryWatchTimer.Start();
+        }
+
+        private void MemoryManager_MemoryChanged(object sender, object e)
+        {
+            UpdateMemoryUsage();
+        }
+
+        private void UpdateMemoryUsage()
+        {
+            var memoryUsed = MemoryManager.AppMemoryUsage;
+            var totalMemory = MemoryManager.AppMemoryUsageLimit;
+            MemoryUsage = SizeSuffix((long)memoryUsed);
+            TotalMemory = SizeSuffix((long)totalMemory);
+        }
+
+        private void Current_SizeChanged(object sender, Windows.UI.Core.WindowSizeChangedEventArgs e) => SetWindowSize();
+
+        private void SetWindowSize()
+        {
+            WindowHeight = Window.Current.Bounds.Height.ToString(); ;
+            WindowWidth = Window.Current.Bounds.Width.ToString();
+        }
+
+        private async void Logger_MessageReceived(object sender, LoggerMessageEventArgs e)
+        {
+            await _semaphoreSlim.WaitAsync();
+
+            var formatedMessage = LogFormatter.GetFormattedLogMessage(e);
+
+            if (AppLogs == null)
+                return;
+
+            await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal,
+            () =>
+            {
+                AppLogs.Add(formatedMessage);
+            });
+
+            _semaphoreSlim.Release();
+        }
+
+        [RelayCommand]
+        private async Task OpenLogFolderAsync()
+        {
+            var logsLocation = LogFormatter.LogFolderPath;
+            var folder = await StorageFolder.GetFolderFromPathAsync(logsLocation);
+
+            await Launcher.LaunchFolderAsync(folder);
+        }
+
+        [RelayCommand]
+        private async Task DeleteUserDataAsync()
+        {
+            _deleteUserDataDialog = new ContentDialog()
+            {
+                Title = "Delete user data",
+                Content = "Are you sure you want to delete user data? If you hit yes, the app will restart.",
+                PrimaryButtonText = "Yes",
+                SecondaryButtonText = "No",
+                PrimaryButtonCommand = PerformUserDataDeletionCommand
+            };
+
+            await _deleteUserDataDialog.ShowAsync();
```

Comment:
Please merge `feature/settings-ui` into this branch, and add a `ShowType` enum in the first parameter. This new extension method allows us to queue or interrupt any `ContentDialog` that's already open, instead of the default behavior of just crashing.

Reply:
Upon testing: QueueNext didn't work as expected 
![image](https://user-images.githubusercontent.com/15306909/209938519-ba4ac8c6-5d4e-45ce-a83c-d3fcb14a6852.png)
![image](https://user-images.githubusercontent.com/15306909/209938584-4f2dd046-70c4-4936-8476-35954f87d664.png)

ShowType Inturrupted works but crashes on retry by saying _inturrupted dialog is null 
`Parameter "_interruptedDialog" (Windows.UI.Xaml.Controls.ContentDialog) must be null, was <Windows.UI.Xaml.Controls.ContentDialog> (Windows.UI.Xaml.Controls.ContentDialog).`
Parameter name: _interruptedDialog
in short both of them have issues
i'll undo my changes for now and let you fix these first